### ISSUE TITLE
Index should contain ISO Date with Timezone Z

### ISSUE THREAD

**Issue by Siedlerchr** [smartsdk/ngsi-timeseries-api#304]
**Is your feature request related to a problem? Please describe.**
When querying history Data, I always get the data back in UTC, however, as the 'Z' identifier is missing, javascript interprets this as local date time. This local time is then currently 2h off (Europe/Berlin) 
It should be consistent with Orion which reports dates back in ISO as:
` 2020-04-07T12:02:07.00Z`

```
"index": [
    "2020-04-07T11:32:09.000",
    "2020-04-07T12:02:07.000",
]
```

**Describe the solution you'd like**
Add the UTC Timezone Identifier "Z" to the data.

**Comment by c0c0n3** [comment]
@Siedlerchr thank you for reporting this :-)

### Implementation notes
The index value can easily be returned in [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) UTC representation by specifying the UTC Z timezone in `_get_isoformat` (Crate translator). That method gets called when formatting query results into the output JSON returned to the client.

But we should dig deeper to review the insert logic too. In fact, the Crate time index is in millis since the epoch and input dates will get converted to Unix time. I remember seeing dates without a timezone in which case I think (to be verified!) QL defaults the timezone to local---that's all we can do really. Perhaps the data I saw didn't come from Orion though...

**Comment by Siedlerchr** [comment]
Thanks for the quick reply. I looked at the python script and it seems like it's quirk of python which does not emit the "Z"
However, for python 3.x there seems to be a solution: https://stackoverflow.com/a/28164131

**Comment by chicco785** [comment]
This is the code for query:
https://github.com/smartsdk/ngsi-timeseries-api/blob/master/src/translators/crate.py#L128

While for injection, i feel this is delegate to crate (can't spot any conversion):
https://github.com/smartsdk/ngsi-timeseries-api/blob/master/src/translators/crate.py#L226

If that's the case, then crate converts any input to epoch using UTC timezone. So, if the input express timezone that will be preserved, if not assumption is that timezone is UTC.

As result, all returned timestamps from crate are indeed UTC, so just appending 'Z' should solve the issue.

**Comment by c0c0n3** [comment]
@chicco785 that's great. Cross-referencing #300 which introduces a fix when generating a time index on insert.

I reckon we should add some tests *after* merging #300 to see what still needs fixing, if anything.

**Comment by chicco785** [comment]
hardening timezone support, is not that simple, but hopefully is covered here: #315

**Comment by c0c0n3** [comment]
@Siedlerchr we've just merged #315 which, among other things, implemented a fix for timezones. Now QuantumLeap returns time values in ISO 8601 format with [offsets from UTC][iso8601.offset]. In the case of UTC time, an offset of `+00:00` gets appended to the time as in the example below

    2020-06-22T01:20:01.567             # before PR 315
    2020-06-22T01:20:01.567+00:00       # after

This isn't the same as [ISO 8601 format with UTC][iso8601.z] that you requested (i.e. trailing `Z`) but is equivalent according to ISO 8601, e.g. the two values below refer to the exact same point in time

    2020-06-22T01:20:01.567+00:00
    2020-06-22T01:20:01.567Z


[iso8601.offset]: https://en.wikipedia.org/wiki/ISO_8601#Time_offsets_from_UTC
[iso8601.z]: https://en.wikipedia.org/wiki/ISO_8601#Coordinated_Universal_Time_(UTC)

**Comment by Siedlerchr** [comment]
Thanks for the fix. That should also work.
